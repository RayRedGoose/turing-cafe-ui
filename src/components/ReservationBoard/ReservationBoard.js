import React from 'react'
import './ReservationBoard.scss'
import ReservationCard from '../ReservationCard/ReservationCard'

const ReservationBoard = ({reservations, removeReserv}) => {

  const cards = reservations.map(reserv => (
    <ReservationCard
      key={reserv.id}
      info={reserv}
      removeReserv={removeReserv}/>
  ))

  return (
    <section className="reserv-board">
      { cards }
    </section>
  )
}

export default ReservationBoard
