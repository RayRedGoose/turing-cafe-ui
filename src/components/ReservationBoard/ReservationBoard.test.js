import React from 'react'
import { shallow } from 'enzyme'
import ReservationBoard from './ReservationBoard'

it('renders with all data passed in correctlly', () => {
  const mockProps = [{
    id: 1,
    name: 'Ray',
    date: '12/12',
    time:'5:00',
    number: '5'
  }]

  const board = shallow(
    <ReservationBoard
      reservations={mockProps}
      removeReserv={jest.fn()}/>
  )

  expect(board).toMatchSnapshot()
})
