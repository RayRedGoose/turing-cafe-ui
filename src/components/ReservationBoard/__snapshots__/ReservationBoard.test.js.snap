// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with all data passed in correctlly 1`] = `
<section
  className="reserv-board"
>
  <ReservationCard
    info={
      Object {
        "date": "12/12",
        "id": 1,
        "name": "Ray",
        "number": "5",
        "time": "5:00",
      }
    }
    key="1"
    removeReserv={[MockFunction]}
  />
</section>
`;
